File: das_backend/src/test/java/ch/sbb/backend/logging/inftrastructure/config/TenantConfigTest.java
package ch.sbb.backend.logging.inftrastructure.config;

import static org.assertj.core.api.Assertions.assertThat;

import ch.sbb.backend.TestContainerConfiguration;
import ch.sbb.backend.logging.domain.model.Tenant;
import ch.sbb.backend.logging.infrastructure.config.TenantConfig;
import java.util.List;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.context.annotation.Import;

@SpringBootTest
@Import(TestContainerConfiguration.class)
class TenantConfigTest {

    @Autowired
    private TenantConfig tenantConfig;

    @Test
    void should_load_tenants_from_configuration() {
        List<Tenant> tenants = tenantConfig.getTenants();
        assertThat(tenants).isNotNull()
            .isNotEmpty()
            .hasSize(1);
        Tenant tenant = tenants.getFirst();
        assertThat(tenant.name()).isEqualTo("test");
        assertThat(tenant.id()).isEqualTo("3409e798-d567-49b1-9bae-f0be66427c54");
        assertThat(tenant.issuerUri()).isEqualTo("https://login.microsoftonline.com/3409e798-d567-49b1-9bae-f0be66427c54/v2.0");
        assertThat(tenant.jwkSetUri()).isEqualTo("https://login.microsoftonline.com/3409e798-d567-49b1-9bae-f0be66427c54/discovery/v2.0/keys");
    }
}

File: backend/src/test/kotlin/ch/sbb/backend/logging/TenantConfigTest.kt
package ch.sbb.backend.logging

import ch.sbb.backend.BaseIT
import ch.sbb.backend.TestContainerConfiguration
import ch.sbb.backend.logging.infrastructure.config.TenantConfig
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Import

@Import(TestContainerConfiguration::class)
class TenantConfigTest: BaseIT() {

    @Autowired
    private lateinit var tenantConfig: TenantConfig

    @Test
    fun `should load tenants from configuration`() {
        val tenants = tenantConfig.tenants
        assertNotNull(tenants)
        assertTrue(tenants.isNotEmpty())
        assertEquals(tenants.size, 1)
        assertEquals(tenants[0].name, "test")
        assertEquals(tenants[0].id, "3409e798-d567-49b1-9bae-f0be66427c54")
        assertEquals(
            tenants[0].issuerUri,
            "https://login.microsoftonline.com/3409e798-d567-49b1-9bae-f0be66427c54/v2.0"
        )
        assertEquals(
            tenants[0].jwkSetUri,
            "https://login.microsoftonline.com/3409e798-d567-49b1-9bae-f0be66427c54/discovery/v2.0/keys"
        )
    }
}
